beard/clustering/wrappers.py

```diff
@@ -151,8 +170,26 @@ def fit(self, X, y=None):
                                       criterion=self.criterion,
                                       depth=self.depth, R=self.R,
                                       monocrit=self.monocrit)
-
-                score = self.scoring(y[train], labels[train])
+                if ground_truth:
+                    train = (y != -1)
+
+                    if not self.affinity_score:
+                        score = self.scoring(y[train], labels[train])
+                    else:
+                        # For scoring function that accepts affinity
+                        score = self.scoring(y[train], labels[train], X)
+
+                elif self.affinity_score:
+                    n_labels = len(np.unique(labels))
+                    n_samples = Xs.shape[0]
+
+                    if 1 < n_labels < n_samples:
+                        score = self.scoring(Xs, labels)
+                    else:
+                        score = -np.inf
+                else:
+                    # could have a scoring function that accepts only labels
+                    return self
```

Comment:
You should keep `b3_f_score` as default scoring, along with `affinity_score=False` by default. It should trigger the correct path, shouldnt it?